src/Models/HydrostaticFreeSurfaceModels/split_explicit_free_surface.jl

```diff
@@ -132,55 +167,128 @@ function SplitExplicitAuxiliary(grid::AbstractGrid)
+    averaging_weights :: ℳ
+    "mass_flux_weights : (`Vector`)"
+    mass_flux_weights :: ℳ
+    "fractional step: (`Number`), the barotropic time step will be (Δτ ⋅ Δt)" 
+    Δτ :: 𝒯
+    "time stepping scheme"
+    timestepper :: 𝒮
 end
 
-function SplitExplicitSettings(; substeps = 200, velocity_weights = nothing, free_surface_weights = nothing)
-    velocity_weights = Tuple(ones(substeps) ./ substeps)
-    free_surface_weights = Tuple(ones(substeps) ./ substeps)
+"""
+    Possible barotropic time-stepping scheme. 
 
-    return SplitExplicitSettings(substeps,
-        velocity_weights,
-        free_surface_weights)
+- `AdamsBashforth3Scheme`: η = f(U, Uᵐ⁻¹, Uᵐ⁻²) then U = f(η, ηᵐ, ηᵐ⁻¹, ηᵐ⁻²)
+- `ForwardBackwardScheme`: η = f(U)             then U = f(η)
+"""
+struct AdamsBashforth3Scheme end
+struct ForwardBackwardScheme end
+
+# Weights that minimize dispersion error from http://falk.ucsd.edu/roms_class/shchepetkin04.pdf (p = 2, q = 4, r = 0.18927)
+@inline function averaging_shape_function(τ; p = 2, q = 4, r = 0.18927) 
+    τ₀ = (p + 2) * (p + q + 2) / (p + 1) / (p + q + 1) 
+    return (τ / τ₀)^p * (1 - (τ / τ₀)^q) - r * (τ / τ₀)
 end
 
-function SplitExplicitSettings(substeps)
-    velocity_weights = Tuple(ones(substeps) ./ substeps)
-    free_surface_weights = Tuple(ones(substeps) ./ substeps)
+@inline averaging_cosine_function(τ) = τ >= 0.5 && τ <= 1.5 ? 1 + cos(2π * (τ - 1)) : 0.0
+
+@inline averaging_fixed_function(τ) = 1.0
 
-    return SplitExplicitSettings(substeps = substeps,
-        velocity_weights = velocity_weights,
-        free_surface_weights = free_surface_weights)
+"""
+    SplitExplicitSettings(; substeps = 200, 
+                            averaging_function = averaging_shape_function,
+                            timestepper = ForwardBackwardScheme())
+
+constructor for `SpliExplicitSettings`. The keyword arguments can be provided
+directly to the `SplitExplicitFreeSurface` constructor
+
+Keyword Arguments
+=================
+
+- `substeps`: The number of substeps that divide the range `(t, t + 2Δt)`. NOTE: not all averaging functions
+              require to substep till `2Δt`. The number of substeps will be reduced automatically to the last index
+              of `averaging_weights` for which `averaging_weights > 0`
+- `averaging_function`: function of `τ` used to average `U` and `η` within the barotropic advancement.
+                        `τ` is the fractional substep going from 0 to 2 with the baroclinic time step `t + Δt`
+                        located at `τ = 1`. This function should be centered at `τ = 1` (i.e. ∑(aₘ⋅m/M) = 1)
+- `timestepper`: Time stepping scheme used, either `ForwardBackwardScheme` or `AdamsBashforth3Scheme`
+"""
+function SplitExplicitSettings(; substeps = 200, 
+                                 averaging_function = averaging_shape_function,
+                                 timestepper = ForwardBackwardScheme())
+
+    τᶠ = range(0.0, 2.0, length = substeps+1)
```

Comment:
```suggestion
    τᶠ = range(0, 2, length = substeps+1)
```

It seems this part of the code will not respect `Float32`. Probably better to build that in now than have to revisit in the future.